sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1190,6 +1190,22 @@ class Column(val expr: Expression) extends Logging {
    */
   def cast(to: String): Column = cast(CatalystSqlParser.parseDataType(to))
 
+  /**
+   * Casts the column to a pretty string.
+   * {{{
+   *   // Casts colA to pretty string.
+   *   df.select(df("colA").toPrettyString())
+   * }}}
+   *
+   * @group expr_ops
+   * @since 3.2.0
+   */
+  def toPrettyString: Column = withExpr {
```

Comment:
Why do we need this API?

Reply:
Add this API  for 
```
if (sparkSession.sessionState.conf.legacyCastToString) {
          Column(col).cast(StringType)
        } else {
          Column(col).toPrettyString
        }
```

Reply:
Shall we make this as a proper SQL expression? e.g.) `to_hive_string`. It makes less sense to expose this API at Column for DSL, that is specific to SQL compatibility.

Reply:
If this is only internal purpose, we can hide it via `private[sql]` or something.

Reply:
> If this is only internal purpose, we can hide it via `private[sql]` or something.

Prefer to do like this. add private[sql]

Reply:
We can construct expressions directly, why do we need an API here?